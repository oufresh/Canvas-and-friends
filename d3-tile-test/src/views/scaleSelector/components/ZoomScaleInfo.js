//@flow
import React from "react";
import styled from "styled-components";

const Container = styled.div`
  font-family: Helvetica Neue, Helvetica, Arial, Liberation Sans, sans-serif;
  height: 35px;
  display: block;
  margin: 1px;
`;

const LineContainer = styled.div`
  width: 160px;
  height: 35px;
  overflow: hidden;
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
`;
const ZoomLabel = styled.div`
  color: rgba(250, 250, 250, 0.6);
  font-size: smaller;
  width: 50px;
`;

const MainLine = styled.div`
  width: 95px;
  height: 1px;
  background-color: rgba(250, 250, 250, 0.6);
`;

const ExtremeLine = styled.div`
  width: 2px;
  height: 6px;
  background-color: rgba(250, 250, 250, 0.6);
`;

const TargetLine = styled.div`
  width: 2px;
  height: 5px;
  background-color: rgba(250, 250, 250, 0.6);
  position: absolute;
  left: ${props => props.left + 40}px;
`;

const TargetText = styled.label`
  color: rgba(250, 250, 250, 0.6);
  position: absolute;
  left: ${props => props.left + 32}px;
  bottom: 22px;
  font-size: smaller;
`;

type Props = {|
  scaleCurrentDecimal: number,
  scaleMin: number,
  scalesSize: number
|};

const ZoomScaleInfo = ({
  scaleCurrentDecimal,
  scaleMin,
  scalesSize
}: Props) => {
  const lineWidth = 97;
  const x1 = 10;
  const x = x1 + (lineWidth / scalesSize) * (scaleCurrentDecimal - scaleMin);
  return (
    <Container>
      <LineContainer id="zoom-line-container">
        <ZoomLabel id="zoom-text-label">ZOOM </ZoomLabel>
        <ExtremeLine id="zoom-first-extreme-line" />
        <MainLine id="zoom-main-line" />
        <ExtremeLine id="zoom-second-extreme-line" />
        <TargetLine id="zoom-target-line" left={x} />
        <TargetText left={x}> {scaleCurrentDecimal.toFixed(1)} </TargetText>
      </LineContainer>
    </Container>
  );
};

export { ZoomScaleInfo };
