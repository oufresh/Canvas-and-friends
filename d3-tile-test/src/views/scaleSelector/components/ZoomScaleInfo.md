### ZoomScaleInfo 5 scalesSize e scaleCurrentDecimal 0

```js
<div style={{ backgroundColor: "black" }}>
  <ZoomScaleInfo
    scalesSize={5}
    scaleMin={-0.49}
    scaleCurrentDecimal={-0.49}
  />
</div>
```

### ZoomScaleInfo 5 scalesSize e scaleCurrentDecimal 1.8

```js
<div style={{ backgroundColor: "black" }}>
  <ZoomScaleInfo
    scalesSize={5}
    scaleMin={-0.49}
    scaleCurrentDecimal={1.8}
  />
</div>
```

### ZoomScaleInfo 5 scalesSize e scaleCurrentDecimal 4.50

```js
<div style={{ backgroundColor: "black" }}>
  <ZoomScaleInfo
    scalesSize={5}
    scaleMin={-0.49}
    scaleCurrentDecimal={4.49}
  />
</div>
```