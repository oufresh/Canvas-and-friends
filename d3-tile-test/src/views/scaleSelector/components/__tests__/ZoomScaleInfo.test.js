import React from "react";
import { shallow } from "enzyme";
import { ZoomScaleInfo } from "../ZoomScaleInfo";

describe("ZoomScaleInfo Component", () => {
  it("renders without a crash ", () => {
    const component = shallow(
      <ZoomScaleInfo
        scalesSize={5}
        scaleMin={-0.49}
        scaleCurrentDecimal={4.49}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
