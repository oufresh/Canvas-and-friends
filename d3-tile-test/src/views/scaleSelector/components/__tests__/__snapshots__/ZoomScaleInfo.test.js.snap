// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomScaleInfo Component renders without a crash  1`] = `
<styled.div>
  <styled.div
    id="zoom-line-container"
  >
    <styled.div
      id="zoom-text-label"
    >
      ZOOM 
    </styled.div>
    <styled.div
      id="zoom-first-extreme-line"
    />
    <styled.div
      id="zoom-main-line"
    />
    <styled.div
      id="zoom-second-extreme-line"
    />
    <styled.div
      id="zoom-target-line"
      left={106.612}
    />
    <styled.label
      left={106.612}
    >
       
      4.5
       
    </styled.label>
  </styled.div>
</styled.div>
`;
